cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java

```diff
@@ -483,6 +484,18 @@ public CloudSecurityGroups securityGroups(CloudCredential cloudCredential, Regio
         return new CloudSecurityGroups(result);
     }
 
+    private List<SecurityGroup> fetchSecurityGroups(AmazonEC2Client ec2Client, DescribeSecurityGroupsRequest describeSecurityGroupsRequest) {
+        try {
+            return ec2Client.describeSecurityGroups(describeSecurityGroupsRequest).getSecurityGroups();
+        } catch (AmazonEC2Exception e) {
+            if (e.getStatusCode() == HttpStatus.BAD_REQUEST.value() || e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
+                throw new IllegalArgumentException(e.getErrorMessage(), e);
```

Comment:
I think this should be IllegalStateException or some other one but definitely not IllegalArgumentException since we don't know whether this method has got some illegal input argument or just something happened.

Reply:
It is just an `internal` exception. I'm fine with a dedicated exception, but the CloudConnectorEception is not the best solution. @bergerdenes if you search in the pr, I handle the exception in `GetPlatformSecurityGroupsHandler`. I need to mark as a Permanent exception and I need to exit from the retry. I think the CloudConnector is a retryable exception in this code